src/batch/src/executor/join/hash_join.rs

```diff
@@ -161,19 +161,19 @@ impl<K: HashKey + Send + Sync> Executor for HashJoinExecutor<K> {
             match take(&mut self.state) {
                 HashJoinState::FirstProbe(probe_table) => {
                     let ret = self.probe(true, probe_table).await?;
-                    if let Some(data_chunk) = ret {
+                    if let Some(data_chunk) = ret && data_chunk.cardinality() > 0 {
```

Comment:
If we assume that child executor should never return 0 cardinality data chunk, we should return an error rather skip it.

Reply:
It's possible for an executor to generate a 0 cardinality chunk, but it just shouldn't output it to the downstream executor. Why consider it as an error.

Reply:
I see, but why we may produce an empty data chunk?

Reply:
One possible situation is inner join with non-equi conditions. In our implementation, we first generate a chunk with equi condition and then filter out result by non-equi condition. If we are unlucky, all result rows are filtered out, the we get and 0 cardinality chunk. This may happen in other executors as well. We should investigate that later.

Reply:
I prefer to implement never return 0 len data chunk. But we can do this later after refactored with stream api.